comps/cores/mega/gateway.py

```diff
@@ -943,15 +990,26 @@ async def handle_request(self, request: Request):
             stream_opt = False
         chat_request = ChatCompletionRequest.model_validate(data)
         # Multimodal RAG QnA With Videos has not yet accepts image as input during QnA.
-        prompt_and_image = self._handle_message(chat_request.messages)
-        if isinstance(prompt_and_image, tuple):
-            # print(f"This request include image, thus it is a follow-up query. Using lvm megaservice")
-            prompt, images = prompt_and_image
-            cur_megaservice = self.lvm_megaservice
-            initial_inputs = {"prompt": prompt, "image": images[0]}
+        messages = self._handle_message(chat_request.messages)
+        decoded_audio_input = ""
+        if isinstance(messages, tuple):
+            prompt, b64_types = messages
+            if "audio" in b64_types:
+                # Audio querys will always come in as a tuple in order to keep track of translation for metadata            
+                decoded_audio_input = b64_types["audio"]
+
+            if "image" in b64_types:
+                # This is an image query, hence it is a follow up query and is using LVM
+                cur_megaservice = self.lvm_megaservice
+                initial_inputs = {"prompt": prompt, "image": b64_types["image"][0]}
+            else:    
+                # This query includes only text, translated audio, or both, hence it will use Embedding
+                cur_megaservice = self.megaservice
+                initial_inputs = {"text": prompt}
+
         else:
-            # print(f"This is the first query, requiring multimodal retrieval. Using multimodal rag megaservice")
-            prompt = prompt_and_image
+            # This is a query with only text, hence it is a first query and is using Embedding
```

Comment:
We can't assume that because it's text only it's a first query. The first query could have matched with something like a podcast clip (which is text only), so follow up queries will still be text only. I had a similar issue when implementing image queries and here's how I'm solving it on [my branch](https://github.com/mhbuehler/GenAIComps/blob/dina/image_query/comps/cores/mega/gateway.py#L971):
```
num_messages = len(data["messages"]) if isinstance(data["messages"], list) else 1
```